static/js/tracking.ts

```diff
@@ -0,0 +1,231 @@
+import { postJson } from "./comm";
+import { isLoggedIn } from "./utils";
+
+// const WAITING_TIME = 5 * 60 * 1000; // 5min in milliseconds
+const WAITING_TIME = 3000; // 3s for testing
+const ELEMENT_TO_TRACK = [
```

Comment:
To make this list easier to manage, what you could also do use use another data attribute on the HTML. For example:

```
<button data-record>
```

Then, in your onClick handler, you can test for the presence of that attribute. You can even use the attribute value to pass in parameters for the tracking, like the name under which the even should be recorded (you could still default to using the `name` or `id` if the attribute doesn't have a value)

```
<button data-record="run-click">
```